tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -3640,118 +3662,134 @@ def test_complex128(a):
-        arr = array([4,5,6,2,1], NumpyInt)
+    @types('int[:]')
+    def test_int(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
     # should be uncommented after resolving #733
-    # def test_int8():
-    #     from numpy import real, shape, array, int8
-    #     arr = array([4,5,6,2,1], int8)
+    # @types('int8[:]')
+    # def test_int8(arr):
+    #     from numpy import real, shape
     #     a = real(arr)
     #     s = shape(a)
     #     return len(s), s[0], a[0]
 
-    # def test_int16():
-    #     from numpy import real, shape, array, int16
-    #     arr = array([4,5,6,2,1], int16)
+    # # @types('int16[:]')
+    # def test_int16(arr):
+    #     from numpy import real, shape
     #     a = real(arr)
     #     s = shape(a)
     #     return len(s), s[0], a[0]
 
-    def test_int32():
-        from numpy import real, shape, array, int32
-        arr = array([4,5,6,2,1], int32)
+    @types('int32[:]')
+    def test_int32(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_int64():
-        from numpy import real, shape, array, int64
-        arr = array([4,5,6,2,1], int64)
+    @types('int64[:]')
+    def test_int64(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_float():
-        from numpy import real, shape, array
-        arr = array([4,5,6,2,1], float)
+    @types('float[:]')
+    def test_float(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_float32():
-        from numpy import real, shape, array, float32
-        arr = array([4,5,6,2,1], float32)
+    @types('float32[:]')
+    def test_float32(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_float64():
-        from numpy import real, shape, array, float64
-        arr = array([4,5,6,2,1], float64)
+    @types('float64[:]')
+    def test_float64(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_complex64():
-        from numpy import real, shape, array, complex64
-        arr = array([4,5,6,2,1], complex64)
+    @types('complex64[:]')
+    def test_complex64(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_complex128():
-        from numpy import real, shape, array, complex128
-        arr = array([4,5,6,2,1], complex128)
+    @types('complex128[:]')
+    def test_complex128(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
+    import numpy as np
```

Comment:
the import of `numpy as np` is already done in this [line ](https://github.com/pyccel/pyccel/pull/763/files#diff-666e38e0289c81a3a42526c953aabe890f5b6e8d0d40fe03e9260900df37d6c0R3551)

Reply:
sorry but i can't see the line.

Reply:
Imports should go at the top of the file unless you have a special reason for them to be elsewhere (e.g. in a function which will be epyccelized to avoid loosing the import)